se/server/pom.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+	<artifactId>kumuluz-server-se</artifactId>
+    <parent>
+        <artifactId>kumuluz-se</artifactId>
+        <groupId>com.kumuluz.ee</groupId>
+        <version>3.6.0-SNAPSHOT</version>
+    </parent>
+
+	<name>KumuluzEE SE server</name>
+	<description>KumuluzEE server for Java SE env</description>
+	
+    <dependencies>
+		<dependency>
+			<groupId>com.kumuluz.ee</groupId>

```

Comment:
Should be `<scope>provided</scope>`

Reply:
Can't really be `provided` because of all the exclusions, do i do it anyway?

Reply:
That's right. Actually it makes no sense to include kumuluzee-cdi-weld dependency here as that is using the Weld servlet version and you are using Weld SE.

I think the right way to go is to move common dependencies to a new kumuluzee-cdi-weld-common module and move your implementation to kumuluzee-cdi-weld-standalone module, since in a way it is a CDI implementation.

Existing interceptors and producers will also need to be moved to the common module, which also makes sense structurally.

The `@EeComponentDependency(EeComponentType.CDI)` will need to be removed from SeServer class. Also a new Component class has to be added with `@EeComponentDef(name = "Weld-SE", type = EeComponentType.CDI)` in order for the dependencies to detect presence of CDI correctly.

I think this structure makes sense. If you have any other suggestions I'm open to them.

Reply:
So, do i understand correctly:
1) add a new component e.g. `kumuluzee-cdi-weld-se` as follows
```
kumuluzee
|-- components
    |-- cdi
        |-- kumuluzee-cdi-weld-se
            |-- weld-se-core
```
2) `kumuluzee-cdi-weld` stays as it is (no reason to modify it really);
3) `kumuluzee-cdi-weld-se` incapsulates `weld-se-core` and
3.1) declares a new `@EeComponentDef` that provides Weld-SE to Kumuluz, essentially becoming an exclusive alternative to `kumuluzee-cdi-weld`;
4) `kumuluz-server-se` depends only on `kumuluzee-cdi-weld-se` and
4.1) `SeServer` still manages `SeContext` as it currently does;
4.2) doesn't declare `@EeComponentDependency` on CDI.

Reply:
>>Existing interceptors and producers will also need to be moved to the common module, which also makes sense structurally.

I see what you mean now, it makes sense indeed. So `kumuluzee-cdi-weld` will only contain the `CdiComponent`, all other stuff will move to `kumuluzee-cdi-weld-common`.

Reply:
Yes, as for the structure I meant the following:

```
kumuluzee
|-- components
    |-- cdi
        |-- kumuluzee-cdi-weld-common (interceptors, producers and common deps)
        |-- kumuluzee-cdi-weld (component class for servlet)
        |-- kumuluzee-cdi-weld-standalone (your implementation)
```

Reply:
I commited the changes :).